src/nvim/window.c

```diff
@@ -2434,29 +2434,10 @@ int win_close(win_T *win, bool free_buf)
   frame_T *win_frame = win->w_floating ? NULL : win->w_frame->fr_parent;
   const bool had_diffmode = win->w_p_diff;
 
-  if (last_window() && !win->w_floating) {
-    EMSG(_("E444: Cannot close last window"));
-    return FAIL;
-  }
-
   if (win->w_closing
       || (win->w_buffer != NULL && win->w_buffer->b_locked > 0)) {
     return FAIL;     // window is already being closed
   }
-  if (win == aucmd_win) {
```

Comment:
`win_close()` is called in a lot of places, let's not move unrelated checks. They are not expected to trigger in "internal" `win_close()` usage, but could catch bug in further changes, potentially.

Reply:
I see, I have moved the checks back to `win_close()` in my last commit.